app/assets/javascripts/pretty-text/engines/discourse-markdown/quotes.js

```diff
@@ -43,7 +37,10 @@ const rule = {
           continue;
         }
 
+        // if we have the additional attribute of username: because we are prioritizing full name
+        // then assign the name to be the displayName
         if (split[i].indexOf("username:") === 0) {
+          displayName = split[0];
           username = split[i].substr(9);
```

Comment:
the substr[9] here is a bit of a concern, we allow infinitely long names but truncate usernames at 9?

Reply:
So the substr(9) is not truncating usernames, it is a selector
```
username:
```
if `username` is present in quote block
```
if (split[i].indexOf("username:") === 0)
```
we select that key/value and remove `username` which is nine characters. This also protects us from users with `,` in their names, because we are not slicing at `,`. We will always have the expected result we desire. See attached test.